src/jit/lowerxarch.cpp

```diff
@@ -2654,45 +2653,29 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode, Ge
                 case NI_Base_Vector128_CreateScalarUnsafe:
                 case NI_Base_Vector256_CreateScalarUnsafe:
                 {
-                    // These intrinsics need to go down this code path even though they
-                    // don't care about the value of the upper bits. This is because we
-                    // can't know for certain that the 32-bit read that would happen
-                    // won't result in an AV.
-
-                    assert(supportsSIMDScalarLoads == false);
-
-                    // These intrinsics require the contained operand to be the same
-                    // size as the baseType for the containing node.
+                    // These intrinsics require the contained operand to be the same size as the widened
+                    // baseType (genActualType(baseType) of the containing node. This is because codegen 
+                    // uses `emitActualTypeSize(baseType)`.
                     //
-                    // If it isn't, we can't mark the node as contained because it
-                    // might avoid a zero or sign-extension that should have otherwise
-                    // occurred.
+                    // If it isn't, we can't mark the node as contained because:
+                    //   1) it might avoid a zero or sign-extension that should have otherwise
+                    //      occurred (i.e. the upper bits of the operand, read as the baseType,
+                    //      may not be the zero/sign-extended value of the actual operand).
+                    //   2) it might lead to an AV if the upper bits are not accessible.
                     //
-                    // We need this even though there are overloads that explicitly
-                    // take byte/sbyte/int16/uint16, since the user may be calling
-                    // the int32/uint32 overload with an explicit cast.
+                    // Note that explicit casts from a wider operand will be reflected in the IR,
+                    // and those will fail the IsContainableMemoryOp check.
                     //
                     // Finally, SIMDScalarLoads aren't supported because these intrinsics
                     // take an actual scalar value, rather than taking a vector and only
-                    // operating on the lower bits.ther than taking a vector and only
                     // operating on the lower bits.
 
-                    const unsigned expectedSize = genTypeSize(containingNode->gtSIMDBaseType);
-                    const unsigned operandSize  = genTypeSize(node->TypeGet());
-
-                    supportsGeneralLoads = (operandSize == expectedSize);
+                    assert(supportsSIMDScalarLoads == false);
 
-                    if (!supportsGeneralLoads && ((expectedSize == 1) || (expectedSize == 2)))
```

Comment:
Why remove this here, but keep it for the general `Scalar` category and the `Insert` intrinsics?

Reply:
I think that was an oversight.

Reply:
Also why remove it in general? This was added due to the particularly bad codegen that the intrinsics expecting a 1 or 2-byte operand were seeing, since the expected size is 1- or 2-bytes, but the operand size (in the case of a stack spilled `LCL_VAR`), was 4-bytes.

Reply:
It seems that those are table-driven, and that the codegen doesn't pay attention to the operand size, so it would seem that `Lowering` needs to know exactly what size of load each of these performs.

Reply:
@tannergooding - since you were out of the office, I was trying to make progress on this since it is blocking @fadimounir. I'm happy to turn this back over to you if you'd like.

Reply:
I'm happy either way. It looks like the only changes you've made from the original are fixing up the comments and remove the above lines from the `CreateScalarUnsafe` intrinsics, is that correct?

To elaborate on the fixes I had done...
* The original issue that the overall PR was trying to fix (https://github.com/dotnet/coreclr/issues/21625), as best as I can tell, was caused for intrinsics where codegen expected `4-bytes`, but the node was getting a `1-byte` or `2-byte` operand.
  * For example, `CreateScalarUnsafe(byte)` when the operand was a `LCL_FLD`, but it needs to generate `movd` which requires the operand to be 4-bytes: https://gist.github.com/tannergooding/d566792973f9e8aaae1741089b4eee87#file-jitdump-txt-L415 vs https://gist.github.com/tannergooding/73ac7cd1ff4a1fb16395f2fae86c9c6b.
  * The fix, which was just doing `supportsGeneralLoads = (operandSize == expectedSize)`, updated these nodes to no longer be contained 
* However, that then led to a number of regressions (especially around the `Create(byte, byte, ..., byte)` methods) for when the codegen expected `1-byte` or `2-bytes`, but the node was getting a stack-spilled operand that was larger
  * These all seemed to be cases where the original operand was 1 or 2-bytes but it was then stack-spilled as a `LCL_VAR`